lib/services/upstart.c

```diff
@@ -391,51 +391,67 @@ upstart_mask_error(svc_action_t *op, const char *error)
     return FALSE;
 }
 
+/*!
+ * \internal
+ * \brief Process the completion of an asynchronous job start, stop, or restart
+ *
+ * \param[in] pending    If not NULL, DBus call associated with request
+ * \param[in] user_data  Action that was executed
+ */
 static void
-upstart_async_dispatch(DBusPendingCall *pending, void *user_data)
+job_method_complete(DBusPendingCall *pending, void *user_data)
 {
     DBusError error;
     DBusMessage *reply = NULL;
     svc_action_t *op = user_data;
 
-    dbus_error_init(&error);
-    if(pending) {
+    // Grab the reply
+    if (pending != NULL) {
         reply = dbus_pending_call_steal_reply(pending);
     }
 
+    // Determine result
+    dbus_error_init(&error);
     if (pcmk_dbus_find_error(pending, reply, &error)) {
 
         /* ignore "already started" or "not running" errors */
+        op->rc = PCMK_OCF_UNKNOWN_ERROR;
+        op->status = PCMK_EXEC_ERROR;
         if (!upstart_mask_error(op, error.name)) {
```

Comment:
all other cases have PCMK_OCF_OK & PCMK_EXEC_DONE - hard to say if we want to point that out right away and shorten the code or rather want to stress that we are taking care of setting those with all cases

Reply:
this was an intentional construction to make the next commit easier -- setting rc/status moves into upstart_mask_error()'s replacement